code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,551 @@
+				if(prob(25))
+					H.set_hallucination(50, 50)
+				if(prob(15))
+					H.vomit()
+				shake_camera(H, 2, 1)
+
+			if(SHUNT_SEVERITY_CRITICAL)
+				to_chat(H, SPAN_DANGER("You feel an overwhelming sense of nausea and vertigo wash over you, your instincts screaming that something is wrong!"))
+				if(prob(50))
+					H.set_hallucination(100, 100)
+				if(prob(45))
+					H.vomit()
+				shake_camera(H, 5, 4)
+
+	for(var/obj/machinery/light/L in SSmachines.machinery) //Fuck with and or break lights.
+		if(!(L.z in ftl_computer.linked.map_z))
+			continue
+		switch(shunt_sev)
+			if(SHUNT_SEVERITY_MINOR)
+				if(prob(15))
+					L.flicker()
+			if(SHUNT_SEVERITY_MAJOR)
+				if(prob(35))
+					L.flicker()
+
+	for(var/obj/machinery/power/apc/A in SSmachines.machinery)
+		if(!(A.z in ftl_computer.linked.map_z))
+			continue
+		switch(shunt_sev)
+			if(SHUNT_SEVERITY_MAJOR)
+				if(prob(15))
+					A.energy_fail(rand(30, 80))
+				if(prob(10))
+					A.overload_lighting(25)
+
+			if(SHUNT_SEVERITY_CRITICAL)
+				if(prob(35))
+					A.energy_fail(rand(60, 150))
+				if(prob(50))
+					A.overload_lighting(50)
+
+/obj/machinery/ftl_shunt/core/proc/handle_spacefloat(var/mob/living/carbon/human/H)
+	if(!H.check_space_footing())
+		 //Flip a coin ...
+		to_chat(H, SPAN_WARNING("Being untethered from a ship entering FTL is a bad idea, but you roll the dice..."))
+		if(prob(50))
+			to_chat(H, SPAN_NOTICE("and win, surviving the energy dancing over your body. Not unharmed, however."))
+			H.apply_damage(300, IRRADIATE, damage_flags = DAM_DISPERSED)
+			return
+		else
+			to_chat(H, SPAN_DANGER("and lose, being ripped apart in a nanosecond by energies beyond comprehension."))
+			if(H.isSynthetic())
+				new /obj/item/remains/robot(get_turf(H))
+			else
+				new /obj/item/remains/human(get_turf(H))
+			qdel(H)
+
+/obj/machinery/ftl_shunt/core/proc/do_sabotage()
+	var/announcetxt
+	var/uhoh = FALSE
+
+	switch(sabotaged)
+		if(SHUNT_SABOTAGE_MINOR)
+			announcetxt = shunt_sabotage_text_minor
+			for(var/mob/living/carbon/human/H in view(7))
+				to_chat(H, SPAN_DANGER("[src] emits a flash of incredibly bright, searing light!"))
+				H.flash_eyes(FLASH_PROTECTION_NONE)
+			empulse(src, 8, 10)
+
+		if(SHUNT_SABOTAGE_MAJOR)
+			announcetxt = shunt_sabotage_text_major
+
+			for(var/mob/living/carbon/human/H in view(7)) //Effect One: scary text.
+				to_chat(H, SPAN_DANGER("[src] hisses and sparks, before coolant lines burst and spew superheated coolant!"))
+
+			explosion(get_turf(src),-1,-1,8,10) //Effect Two: blow the windows out.
+
+			for(var/obj/machinery/power/apc/A in SSmachines.machinery) //Effect Three: shut down power across the ship.
+				if(!(A.z in ftl_computer.linked.map_z))
+					continue
+				A.energy_fail(rand(60,80))
+
+		if(SHUNT_SABOTAGE_CRITICAL)
+			announcetxt = shunt_sabotage_text_critical
+
+			uhoh = TRUE //let the fuckening begin!
```

Comment:
You don't set `uhoh` anywhere else, why not just spawn the singularity at the end of this block?